### DISCUSSION TITLE
Could this package be compatible with Laravel Telescope?

### DISCUSSION THREAD

**Discussion by koAndy** [jenssegers/laravel-mongodb#1648] [Ideas]
Could this package be compatible with Laravel Telescope?

I tried install Telescope and on dashboard, when trying to read requests, in console I get "Call  to a member function prepare() on null".

**Comment by staf** [comment]
To make Telescope work with this "all" you need is an adapter for the storage driver. Telescope registers the its database driver as part of the service provider, which is where you can replace the bindings to a driver that uses this package instead of eloquent to store the records.

Not sure if anyone have written this yet though, but implementing those interfaces should be somewhat straightforward.

**Comment by yuvraj993** [comment]
I am using mongodb with jenssegers/mongodb package.

exception: "Symfony\Component\Debug\Exception\FatalThrowableError"
file: "/var/www/html/broc/vendor/laravel/framework/src/Illuminate/Database/Connection.php"
line: 326
message: "Call to a member function prepare() on null"

**Reply by LucasSantosDev** [reply]
@yuvraj993 some solution?

**Reply by suracekumar632** [reply]
@yuvraj993  i have same issue any solution ?

**Comment by anthonyvancauwenberghe** [comment]
you mean log the mongodb queries? or store the telescope logs in mongodb?

The first is easy you need to enable the query log:
`DB::connection('mongodb')->enableQueryLog();`
the second is harder. Its probably more than writing a "storage adaptor" since the package might use some raw sql queries :s

Worst case scenario you might have to fork the package & dive into the code

**Comment by solofeed** [comment]
Bump

**Comment by bytewoopy** [comment]
Has anyone been able to get this to work? It is unfortunate to not have this when switching over to mongo. It was great with mysql. I would love to see some examples of what needed to be changed to get it working. It's been really hard to find anything informative online so far.

**Comment by bisht2050** [comment]
Tracked with [PHPORM-168 ](https://jira.mongodb.org/browse/PHPORM-168).
